### PR TITLE
Feature 673 release_guide_met

### LINKED ISSUE #673: Internal: Create a METplus "Release Guide" describing how to build releases for the METplus components.
## Describe the Task ##
Please see [METplus meeting notes](https://docs.google.com/document/d/1nkAPFijlJq3X3vuOBVBPgVSzTn6IUMEAO_gnEaLskyg/edit#) from 10/27/2020 for more details. This task is to create a 3rd document for METplus in addition to the user and contributor's guides. This document will describe the process for creating a METplus release and consist of up to 8 chapters:
1. Introduction
2. METplus Release
3. MET Release
4. METdatadb Release
5. METcalcpy Release
6. METplotpy Release (or combine METcalcpy/METplotpy into a single chapter if that makes sense?)
7. METviewer Release
8. METexpress Release

The docs/Release_Guide directory will contain .rst files for each of the items listed above. It should also contain a subdirectory named "release_steps". Each top-level .rst file should be very sparse and consist basically of 3 bulleted lists:
- official release (e.g. vX.Y)
- bugfix release (e.g. vX.Y.Z)
- development release (e.g. vX.Y-betaN or vX.Y-rcN)

Each element in that list should source on .rst files that live in the "release_steps" sub-directory. It'd be cool if that content could be hidden by default, but click on it to expand it. So while we have separate lists for each component, they serve up common content from the "release_steps" directory.

Migrate content from:
Existing wiki page:
   https://github.com/dtcenter/METplus/wiki/Creating-a-new-METplus-release
Creating met-10.0_beta1 release:
   https://github.com/dtcenter/MET/issues/1523#issuecomment-714781834

### Time Estimate ###
4-8 hours for many different people.

### Sub-Issues ###
Consider breaking the task down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*
May need to break this down, but not quite sure how to do so yet.

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2702691

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required: George (METplus), John HG (MET), Minna (METplotpy/METcalcpy), Tatiana (METviewer), Julie (docs infrastructure), Venita (METdatadb) and Molly (METexpress)
- [x] Select **scientist(s)** or **no scientist** required: Keith (review and advise)

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [x] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)
Details for each component will live in the common METplus Release Guide. So all changes will be in the METplus repo.

## Task Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by JohnHalleyGotway**
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>
Adding first draft of release steps for MET along with consistency updates.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Read through 3 versions of release steps for MET. Consider whether "release_steps/met/create_release_on_github.rst" is general enough to be applied to the other METplus repos.

- [x] Do these changes include sufficient documentation and testing updates? **[Yes]**
That's what they are!

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Release_Guide/release_steps/met/create_release_on_github.rst [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-- Choose a release name (i.e. |projectRepo|-X.Y.Z, |projectRepo|-X.Y.Z-betaN, or |projectRepo|-X.Y.Z)
-- Add a link to the release notes from the |projectRepo| User's Guide
-- Attach a PDF of the |projectRepo| User's Guide, if available
+* Go to Releases on GitHub and click 'Draft a new release'
+
+* Set the tag based on the version, starting with 'v', i.e. vX.Y.Z-betaN, vX.Y.Z-rcN, or vX.Y.Z
+
+* Set the branch value to 'develop' for a development release or 'main_vX.Y' for a bugfix or official release
+
+* Choose a release name (i.e. |projectRepo|-X.Y.Z, |projectRepo|-X.Y.Z-betaN, or |projectRepo|-X.Y.Z)
```
Comment (jprestop) [inline docs/Release_Guide/release_steps/met/create_release_on_github.rst]:
```suggestion
* Choose a release name (i.e. |projectRepo|-X.Y.Z-betaN, |projectRepo|-X.Y.Z-rcN, or |projectRepo|-X.Y.Z)
```
I noticed that the first and last items in "i.e" were the same and was wondering if it should be as I suggest here (betaN, rcN, empty)?

File: docs/Release_Guide/release_steps/met/update_version_development.rst [hunk 2]
```diff
@@ -1,22 +1,18 @@
-Update Development Release Version Number
------------------------------------------
+Update Version Number
+---------------------
 
-*  Work on a feature branch from the *develop* branch being sure to include
-   the GitHub issue number for the new release
-   (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y-rcN).
+*  Create a feature branch from the *develop* branch being sure to include the GitHub issue number for the new release (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y-rcN).
```
Comment (jprestop) [inline docs/Release_Guide/release_steps/met/update_version_development.rst]:
```suggestion
*  Create a feature branch from the *develop* branch being sure to include the GitHub issue number for the new release (e.g. feature_NNNN_vX.Y.Z-betaN or feature_NNNN_vX.Y.Z-rcN).
```
Should the rcN release have a Z also?

File: docs/Release_Guide/release_steps/met/update_version_official.rst [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+  
+  * If necessary, update the version number listed in the MET configuration files:
+
+    * Default configuration files in *met/data/config*.
+
+    * Sample configuration files in *met/scripts/config*.
+
+    * Test configuration files, searching recursively, in *test/config*.
+
+  * If necessary, add a new *met/data/table_files/met_header_columns_VX.Y.txt* defining the columns names for this version.
```
Comment (jprestop) [inline docs/Release_Guide/release_steps/met/update_version_official.rst:26]: Should met_header_columns_VX.Y.txt have a ".Z" or does it stay "X.Y"?

File: docs/Release_Guide/release_steps/met/update_version_official.rst [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
+
+  * If necessary, add a new *test/hdr/met_X_Y.hdr* file defining the column names for this version for the test scripts.
```
Comment (jprestop) [inline docs/Release_Guide/release_steps/met/update_version_official.rst:28]: Should met_X_Y.hdr*  have a".Z" or does it stay "X.Y"?

File: docs/Release_Guide/release_steps/met/update_version_official.rst (hunk 3 above)
Comment (JohnHalleyGotway) [reply docs/Release_Guide/release_steps/met/update_version_official.rst:26]: Yes, X.Y is correct. The headers do not change for bugfix releases... so we only need to keep track of X.Y, not X.Y.Z.

File: docs/Release_Guide/release_steps/met/update_version_official.rst (hunk 4 above)
Comment (JohnHalleyGotway) [reply docs/Release_Guide/release_steps/met/update_version_official.rst:28]: Yes, X.Y is correct. The headers do not change for bugfix releases... so we only need to keep track of X.Y, not X.Y.Z.

**Review by jprestop — APPROVED** [review]
Changes are clear and look great!

**Pull request merged**